gobblin-metastore/src/main/java/gobblin/metastore/FsStateStore.java

```diff
@@ -174,11 +172,13 @@ public void putAll(String storeName, String tableName, Collection<T> states)
     } finally {
       closer.close();
     }
+
+    Path tablePath = new Path(new Path(this.storeRootDir, storeName), tableName);
+    HadoopUtils.renamePath(this.fs, tmpTablePath, tablePath);
```

Comment:
I have the same concerns as @kadaan . @zliu41 what are all of the uses of this method? and for `putAll`?